### PR TITLE
BUG: fix qwen chat3.5 

### LINKED ISSUE #3600: Qwen3.5 Image Lora post-training
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Currently, I am testing the scripts for qwen3.5 finetuning for images. The model merged with Lora should be returning text content.

### Current behaviour

There is no content in the text reply, when I query in image+text or text inputs. I checked it with the base Qwen3.5 to see if its an error in my config but it returns the expected text for given images and text.

### Steps to reproduce

1. Run 10% of epoch of the provided qwen3.5 script https://github.com/axolotl-ai-cloud/axolotl/blob/main/examples/qwen3.5/9b-lora-vision.yaml
2. Test the model after merging the lora adapters on any text prompts or image+text prompts

### Config yaml

```yaml

```

### Possible solution

_No response_

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.12

### axolotl branch-commit

main/e079cf1

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

### PR DESCRIPTION
**Pull request by ved1beta**
# Description
Fine-tuning Qwen3.5 vision models with chat_template: qwen3_5 on datasets whose assistant turns use list/parts content (e.g. HuggingFaceH4/llava-instruct-mix-vsft, OpenAI-style
  [{"type":"text","text":...}]) produces a model that returns empty replies after merge. 

## Motivation and Context
 Fixes #3600. 

## How has this been tested?

repro fixed 

## AI Usage Disclaimer
claude helped with dignose 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved rendering of assistant messages in chat templates to ensure proper text concatenation and display.

### CODE REVIEW

File: src/axolotl/utils/chat_templates/templates/qwen3_5.jinja
Comment (NanoCode012) [inline src/axolotl/utils/chat_templates/templates/qwen3_5.jinja]:
```suggestion
{%- set reasoning_content = content.split('</think>')[0].rstrip('\n').split('<think>')[-1].lstrip('\n') %}
{%- set content = content.split('</think>')[-1].lstrip('\n') %}
```

reasoning_content would be empty since we strip it out first in content when it should be other way around. L97-98

Ref: this is inverted upstream too https://huggingface.co/Qwen/Qwen3.5-397B-A17B/blob/main/chat_template.jinja#L95-L96

ps; not sure I'm selecting the suggestion lines correctly

**Review by NanoCode012 — APPROVED** [review]
